cmd/tools/gen-accessors.go

```diff
@@ -0,0 +1,344 @@
+//+build ignore
+
+/*
+ * Datadog API for Go
+ *
+ * Please see the included LICENSE file for licensing information.
+ *
+ * Copyright 2017 by authors and contributors.
+
+ gen-accessors generates GetXx, GetXxOk, HasXx and SetXx methods for struct field types with pointers.
+
+ It should be run by the go-datadog-api authors to ge-generate methods before pushing changes to GitHub.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+)
+
+const (
+	fileSuffix = "-accessors.go"
+)
+
+var (
+	verbose    = flag.Bool("v", false, "Print verbose log messages")
+	sourceTmpl = template.Must(template.New("source").Parse(source))
+)
+
+func logf(fmt string, args ...interface{}) {
+	if *verbose {
+		log.Printf(fmt, args...)
+	}
+}
+
+func main() {
+	flag.Parse()
+	fset := token.NewFileSet()
+
+	pkgs, err := parser.ParseDir(fset, ".", sourceFilter, 0)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+
+	for pkgName, pkg := range pkgs {
+		t := &templateData{
+			Year:     time.Now().Year(),
+			Filename: pkgName + fileSuffix,
+			Package:  pkgName,
+			Imports:  map[string]string{},
+		}
+		for filename, f := range pkg.Files {
+			logf("Processing %v...", filename)
+			t.sourceFile = filename
+			if err := t.processAST(f); err != nil {
+				log.Fatal(err)
+			}
+		}
+		if err := t.dump(); err != nil {
+			log.Fatal(err)
+		}
+	}
+	logf("Done.")
+}
+
+// processAST walks source code files, looks for struct definitions, and calls appropriate helper functions
+// to add generate accessors for type fields are star expressions -ie: pointers
+func (t *templateData) processAST(f *ast.File) error {
```

Comment:
This is a much fancier approach than the one I had for sure! I think ATM having top-level only is fine, and we can see how things develop.